lib/util/git/index.ts

```diff
@@ -564,18 +568,27 @@ export async function isBranchStale(branchName: string): Promise<boolean> {
 }
 
 export async function isBranchModified(branchName: string): Promise<boolean> {
-  await syncGit();
-  // First check cache
-  if (config.branchIsModified[branchName] !== undefined) {
-    return config.branchIsModified[branchName];
-  }
   if (!branchExists(branchName)) {
     logger.debug(
       { branchName },
       'Branch does not exist - cannot check isModified'
     );
     return false;
   }
+  // First check local config
+  if (config.branchIsModified[branchName] !== undefined) {
+    return config.branchIsModified[branchName];
+  }
+  // Second check repoCache
+  const isModified = getCachedModifiedResult(
+    branchName,
+    config.branchCommits[branchName]
+  );
+  if (!is.null_(isModified)) {
```

Comment:
```suggestion
  if (isModified !== null) {
```
we should use normal null check here

Reply:
Sure. Could you please explain the reason?